airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/state/StreamManager.kt

```diff
@@ -152,8 +152,21 @@ class DefaultStreamManager(
         val expanded =
             batch.ranges.asRanges().map { it.span(Range.singleton(it.upperEndpoint() + 1)) }
 
-        stateRanges.addAll(expanded)
-        log.info { "Updated ranges for ${stream.descriptor}[${batch.batch.state}]: $stateRanges" }
+        when (batch.batch.state) {
+            Batch.State.PERSISTED -> {
+                rangesState[Batch.State.PERSISTED]?.addAll(expanded)
+            }
+            Batch.State.COMPLETE -> {
+                // A COMPLETED state implies PERSISTED, so also mark PERSISTED.
+                rangesState[Batch.State.PERSISTED]?.addAll(expanded)
+                rangesState[Batch.State.COMPLETE]?.addAll(expanded)
+            }
+            else -> Unit
+        }
+
+        log.info {
+            "Updated ranges for ${stream.descriptor}[${batch.batch.state}]: $expanded. PERSISTED is also updated on COMPLETE."
```

Comment:
Not needed for this PR, but a good improvement might be to log the whole table on update.